### DISCUSSION TITLE
Redrawing the waveform with new peaks, new Audiobuffer?

### DISCUSSION THREAD

**Discussion by Slatvinskyi** [katspaugh/wavesurfer.js#3787] [Q&A] [answered]
Hi, i would like ask is it possible to redraw the waveform with new peaks after it was already initialized without destroying the Wavesurfer instance?

I tried to re-render it this way:

wavesurfer.setOptions({
    peaks: [peaks],
 })

It changes other options, color, width, but peaks remain the same.

Thanks.

**Comment by katspaugh** [comment, marked as answer]
You can "load" new peaks like this:

```
wavesurfer.load(url, peaks, duration)
```

**Reply by Slatvinskyi** [reply]
Thank you! That was very helpful.
